docs/usage/integration.md

```diff
@@ -0,0 +1,413 @@
+    (excluding balance fields, which are modified by transfers).
+  - [Transfers](../reference/transfers.md) are immutable — once created, they are never modified.
+  - Transfer [timeouts](../reference/transfers.md#timeout) are deterministic, driven by the
+    cluster's timestamp.
+  - Object timestamps are cluster-unique. For objects `A` and `B`, `A.timestamp ≠ B.timestamp`.
+  - Object timestamps are strictly increasing For objects `A` and `B`, if
+    `A.timestamp < B.timestamp`, then `A` was committed earlier than `B`.
+
+### Reply Order
+
+Replies to a client session always arrive in order — a client session may have only one request
+in-flight, and clients ignore (duplicate) replies to their prior requests.
+
+- Requests are executed in the order they arrive at the cluster's primary.
+- Replies to different clients may arrive out of order.
+
+#### Example
+
+Consider two clients `A` and `B`:
+
+  1. Client `A` sends request `A₁`.
+  2. Client `B` sends request `B₁`.
+
+Client `A` sent its request first, but requests `A₁` and `B₁` may execute in either order —
+whichever arrives first at the primary will execute first.
+
+In this diagram, the requests are delivered out of order — `B₁` then `A₁`:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant Client A
+    participant Client B
+    participant (Network)
+    participant Cluster
+    Client A->>(Network): A₁ (request)
+    Client B->>(Network): B₁ (request)
+    Note over (Network): Request A₁ is delayed by the network.
+    (Network)->>Cluster: B₁ (request)
+    (Network)->>Cluster: A₁ (request)
+    Cluster->>(Network): B₁ (reply)
+    Cluster->>(Network): A₁ (reply)
+    (Network)->>Client B: B₁(reply)
+    (Network)->>Client A: A₁ (reply)
+```
+
+Suppose instead `A₁` arrives and executes before `B₁`.
+The replies may be delivered in the same order (`A₁` then `B₁`), or they may be reordered, as shown below:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant Client A
+    participant Client B
+    participant (Network)
+    participant Cluster
+    Client A->>(Network): A₁ (request)
+    Client B->>(Network): B₁ (request)
+    (Network)->>Cluster: A₁ (request)
+    (Network)->>Cluster: B₁ (request)
+    Cluster->>(Network): A₁ (reply)
+    Cluster->>(Network): B₁ (reply)
+    Note over (Network): Reply A₁ is delayed by the network.
+    (Network)->>Client B: B₁(reply)
+    (Network)->>Client A: A₁ (reply)
+```
+
+### Retries
+
+A client will retry a request until either:
+
+- the client receives a corresponding reply from the cluster, or
+- the client is terminated.
+
+Unlike most database or RPC clients:
+
+- the TigerBeetle client will never time out
+- the TigerBeetle client has no retry limits
+- the TigerBeetle does not surface network errors
+
+With TigerBeetle's strict consistency model, surfacing these errors at the client/application level
+would be misleading, because they would imply that a request did not execute, when that is not known:
```

Comment:
Long sentence could be broken (if you want): `would be misleading. They would imply`

Reply:
And although strictos don't like it, it is actually defensible to start a sentence with "And" or "Because" :D

Reply:
ehh nevermind that would make it not a sentence. Yeah if you break it up you could just drop `Because`

Reply:
Breaking it up is a good suggestion, I'll do:

> With TigerBeetle's strict consistency model, surfacing these errors at the client/application level
> would be misleading. An error would imply that a request did not execute, when that is not known:

Reply:
looks good!